docker/backstopjs/Dockerfile

```
@@ -1,28 +1,33 @@
-FROM alpine:3.17
+FROM node:14.21.1-bullseye
 
-ARG BACKSTOPJS_VERSION=6.1.1
+# Prevent intractive prompts from apt-get.
+ARG DEBIAN_FRONTEND=noninteractive
 
+ARG BACKSTOPJS_VERSION=6.1.1
 ENV BACKSTOPJS_VERSION=$BACKSTOPJS_VERSION
 
-RUN apk add --no-cache \
-      chromium \
-      nss \
-      freetype \
-      harfbuzz \
-      ca-certificates \
-      ttf-freefont \
-      nodejs \
-      npm \ 
-      openrc \
-      dbus
-
-ENV PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium-browser
-
-RUN apk upgrade --no-cache --available && \
-    PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true npm i -g backstopjs@${BACKSTOPJS_VERSION} && \
+RUN apt-get update
+
+# Install suggested requirements for running backstop in GitHub actions.
+# https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#running-puppeteer-on-gitlabci
+RUN apt-get install -yq gconf-service libasound2 libatk1.0-0 libc6 libcairo2 \
+    libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgbm1 libgcc1 libgconf-2-4 \
+    libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 \
+    libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 \
+    libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 \
+    libxss1 libxtst6 ca-certificates fonts-liberation libnss3 lsb-release \
+    xdg-utils wget
+
+# Install the Chromium requirements, specify a Chromium build for stability.
+RUN apt-get install -yq \
+    chromium=120.0.6099.224-1~deb11u1 \
+    ca-certificates \
+    nodejs \
+    npm
```

@benbowler it's worth noting that if we are using a base Docker image with Node preinstalled we shouldn't need to install these dependencies:

```suggestion
    ca-certificates
```

That aside, thinking about the switch from Alpine to Debian, I still do have some reservations on that front if it's going to slow down the setup process and take up more space for the image.

I wonder, if someone has an existing image which was built when things had recently stabilised, we could take a look at the versions of the unpinned dependencies installed on that and compare it to a freshly built image to see what if anything has changed. That might allow us to simply specify some versions in the current Dockerfile for a fairly minimal change... I also wonder if the Alpine patch version could have changed?

---

Of course, that's assuming things _had_ stabilised due to the recently applied fix, it's a bit hard to be sure although it certainly looked like it...

---

One more thing, I don't think we need to specify Node 14 when building the Backstop image, as we're not running our own scripts on it, it just needs Node for running Backstop in the container...

---

Good point here. I've successfully run the Debian version of this today 6 times without issues.